components/availability/src/Availability.svelte

```diff
@@ -273,7 +300,7 @@
       newCalendarTimeslotsForGivenEmails = await getAvailability();
     }
     // When dates_to_show is updated, update availability
-    if (email_ids?.length && dates_to_show) {
+    if (email_ids?.length && dates_to_show && show_as_week !== undefined) {
```

Comment:
This ensures we'll always make a getAvailability() request when show_as_week changes.